config-processing-changelog/2019-10-10.md

```diff
@@ -0,0 +1,76 @@
+# Upcoming changes to v2.1 config processing
+October 10, 2019
+
+Config processing is an area where small changes can have a large impact across
+our customer base. We do our best, through extensive unit, functional, and
+regression testing to make sure that we don't break config processing. But
+sometimes we find bugs where the implementation is not conforming to
+specification and our regression testing tells us that there are people who
+will be affected by the bug fixes.
+
+* Fixed a bug where a `false` default value for a parameter would prevent the
+  default value being type-checked against the parameter's declared type.
+  Parameter declarations like the following will no longer be accepted.
+  ```yaml
+  jobs:
+    my-job:
+      parameters:
+        foo:
+          type: string
+          default: false
+      steps:
+        - ...
+  ```
+  An error will be reported instead:
+  ```
+  Error in definition of job 'my-job'
+  Parameter error: default value of parameter 'foo' is 'false' (type boolean): expected type string
+  ```
+
+* Fixed a bug where a `null` default value for a parameter were accepted as
+  valid defaults. Parameters declared with a `null` default value will always
+  need to be passed in invocations, the following will no longer be accepted:
+  ```yaml
+  jobs:
+    my-job:
+      parameters:
+        foo:
+          type: string
+          default:
```

Comment:
is `default: ""` still valid? and will that evaluate as falsey?

Reply:
```clojure
user> (some? "")
true
``` so `default: ""` should work just fine.

Reply:
Only `null` values for `default` are affected, as documented.

Reply:
Only `null` values for `default` are affected, as documented.